nilearn/connectome/tests/test_connectivity_matrices.py

```diff
@@ -306,40 +442,57 @@ def test_geometric_mean_properties():
     tol = 1e-20
     with warnings.catch_warnings(record=True) as w:
         warnings.simplefilter("always")
-        gmean = _geometric_mean(spds, max_iter=max_iter, tol=tol)
+        _geometric_mean(spds, max_iter=max_iter, tol=tol)
         assert len(w) == 1
+
     grad_norm = grad_geometric_mean(spds, max_iter=max_iter, tol=tol)
+
     assert len(grad_norm) == max_iter
     assert grad_norm[-1] > tol
 
-    # Evaluate convergence. A warning is printed if tolerance is not reached
-    for p in [0.5, 1.0]:  # proportion of badly conditioned matrices
-        spds = [
-            random_spd(n_features, eig_min=1e-2, cond=1e6, random_state=0)
-            for _ in range(int(p * n_matrices))
-        ]
-        spds.extend(
-            random_spd(n_features, eig_min=1.0, cond=10.0, random_state=0)
-            for _ in range(int(p * n_matrices), n_matrices)
-        )
-        max_iter = 30 if p < 1 else 60
-        gmean = _geometric_mean(spds, max_iter=max_iter, tol=1e-5)
+
+# proportion of badly conditioned matrices
+@pytest.mark.parametrize("p", [0.5, 1.0])
+def test_geometric_mean_properties_evaluate_convergence(p):
+    n_matrices = 40
+    n_features = 15
+    # A warning is printed if tolerance is not reached
+    spds = [
+        random_spd(n_features, eig_min=1e-2, cond=1e6, random_state=0)
+        for _ in range(int(p * n_matrices))
+    ]
+    spds.extend(
+        random_spd(n_features, eig_min=1.0, cond=10.0, random_state=0)
+        for _ in range(int(p * n_matrices), n_matrices)
+    )
+    max_iter = 30 if p < 1 else 60
+
+    _geometric_mean(spds, max_iter=max_iter, tol=1e-5)
 
 
 def test_geometric_mean_errors():
     n_features = 5
 
     # Non square input matrix
     mat1 = np.ones((n_features, n_features + 1))
-    pytest.raises(ValueError, _geometric_mean, [mat1])
+
+    with pytest.raises(ValueError, match="Expected a square matrix"):
+        _geometric_mean([mat1])
```

Comment:
I would do three different tests here, one for non square matrices, one for non spd matrices, and one for different shape matrices.